### PR TITLE
Issue #109

### PR DESCRIPTION
**Pull request by autumnwoz**
Basic cool-down for #109

### CODE REVIEW

File: Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java [hunk 1]
```diff
@@ -267,6 +273,24 @@ public int getSpellCount (Spells spell)
       return count;
    }
 
+   /**
+    * Get the casting count for a spell
+    *
+    * @param spell the spell to get a count for
+    * @return the number of times a player has cast this spell
+    */
+   public Long getSpellLastCast (Spells spell)
```
Comment (Azami7) [inline Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java]: Should rename this to be getSpallLastCastTime to be clearer

File: Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java [hunk 2]
```diff
@@ -307,6 +338,39 @@ public void setSpellCount (Spells spell, int count)
       }
    }
 
+   /**
+    * Set the most recent cast time for a spell. This will override the existing values for this spell.
+    *
+    * @param spell the spell to set the time for
+    */
+   public void setSpellRecent (Spells spell)
```
Comment (Azami7) [inline Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java]: maybe setSpellLastCastTime or setSpellCastTime to be clearer?

File: Ollivanders/src/net/pottercraft/Ollivanders2/Spell/SpellProjectile.java [hunk 3]
```diff
@@ -298,4 +298,11 @@ public void kill ()
    {
       kill = true;
    }
+
+   /**
+    * Return the cool-down time for this spell.
+    */
+   public Long getCoolDown() {
+      return new Long(1000);
```
Comment (tcubansk) [inline Ollivanders/src/net/pottercraft/Ollivanders2/Spell/SpellProjectile.java]: public static final Long DEFAULT_COOLDOWN = new Long(1000);

public Long getCoolDown() {return DEFAULT_COOLDOWN; }

File: Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java [hunk 4]
```diff
@@ -307,6 +336,39 @@ public void setSpellCount (Spells spell, int count)
...
+            recentSpells.replace(spell, System.currentTimeMillis() + s.getCoolDown());
+         }
+         else
+         {
+            recentSpells.put(spell, System.currentTimeMillis() + s.getCoolDown());
+         }
+      }
+      catch (InvocationTargetException e)
+      {
+         e.getCause().printStackTrace();
```
Comment (tcubansk) [inline Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java]: Is there a logging framework setup yet?

File: Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java [hunk 5]
```diff
@@ -307,6 +336,39 @@ public void setSpellCount (Spells spell, int count)
...
    }
 
+   /**
+    * Set the most recent cast time for a spell. This will override the existing values for this spell.
+    *
+    * @param spell the spell to set the time for
+    */
+   public void setSpellRecentCastTime (Spells spell)
+   {
+      String spellClass = "net.pottercraft.Ollivanders2.Spell." + spell.toString();
```
Comment (tcubansk) [inline Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java]: In general I would like to see a better way of getting the spell than doing string concatenation and reflection.  Problems with the spellClass string not matching the classname won't show up until runtime when the specific spell is attempted.

**Pull request merged**